extensions/cornerstone/src/services/SyncGroupService/SyncGroupService.ts

```diff
@@ -29,48 +45,71 @@ export default class SyncGroupService {
     Object.assign(this, pubSubServiceInterface);
   }
 
-  private _createSynchronizer(type: string, id: string) {
-    type = type.toLowerCase();
-    if (type === POSITION) {
-      return synchronizers.createCameraPositionSynchronizer(id);
-    } else if (type === VOI) {
-      return synchronizers.createVOISynchronizer(id);
+  private _createSynchronizer(type: string, id: string, options) {
+    const syncCreator = this.synchronizerCreators[type.toLowerCase()];
+    if (syncCreator) {
+      return syncCreator(id, options);
+    } else {
+      console.warn('Unknown synchronizer type', type, id);
     }
   }
 
+  /**
+   * Creates a synchronizer type.
+   * @param type is the type of the synchronizer to create
+   * @param creator
+   */
+  public setSynchronizer(type: string, creator: SyncCreator): void {
+    this.synchronizerCreators[type] = creator;
+  }
+
+  protected _getOrCreateSynchronizer(
+    type: string,
+    id: string,
+    options?: Record<string, unknown>
+  ): Synchronizer {
+    let synchronizer = SynchronizerManager.getSynchronizer(id);
+
+    if (!synchronizer) {
+      synchronizer = this._createSynchronizer(type, id, options);
+    }
+    return synchronizer;
+  }
+
   public addViewportToSyncGroup(
     viewportId: string,
     renderingEngineId: string,
-    syncGroups?: SyncGroup[]
+    syncGroups?: (SyncGroup | string)[]
```

Comment:
This is just a convenience, as most of the time it is sufficient to just label the type.